src/main/java/org/opentripplanner/routing/algorithm/mapping/AlertToLegMapper.java

```diff
@@ -142,59 +151,52 @@ private static void addTransitAlertPatchesToLeg(
   }
 
   private Collection<TransitAlert> getAlertsForStopAndRoute(Stop stop, FeedScopedId routeId) {
-    return getAlertsForStopAndRoute(stop, routeId, true);
-  }
-
-  private Collection<TransitAlert> getAlertsForStopAndRoute(
-    Stop stop,
-    FeedScopedId routeId,
-    boolean checkParentStop
-  ) {
     if (stop == null) {
       return new ArrayList<>();
     }
     Collection<TransitAlert> alertsForStopAndRoute = transitAlertService.getStopAndRouteAlerts(
       stop.getId(),
       routeId
     );
-    if (checkParentStop) {
-      if (alertsForStopAndRoute == null) {
-        alertsForStopAndRoute = new HashSet<>();
+    if (alertsForStopAndRoute == null) {
+      alertsForStopAndRoute = new HashSet<>();
+    }
+
+    if (stop.isPartOfStation()) {
+      //Also check parent
+      final Station parentStation = stop.getParentStation();
+      Collection<TransitAlert> alerts = transitAlertService.getStopAndRouteAlerts(
+        parentStation.getId(),
+        routeId
+      );
+      if (alerts != null) {
+        alertsForStopAndRoute.addAll(alerts);
       }
 
-      if (stop.isPartOfStation()) {
-        //Also check parent
-        Collection<TransitAlert> alerts = transitAlertService.getStopAndRouteAlerts(
-          stop.getParentStation().getId(),
-          routeId
-        );
-        if (alerts != null) {
-          alertsForStopAndRoute.addAll(alerts);
+      // ...and siblings - platform may have been changed
+      for (var siblingStop : parentStation.getChildStops()) {
+        if (!stop.getId().equals(siblingStop.getId())) {
+          Collection<TransitAlert> siblingAlerts = transitAlertService.getStopAndRouteAlerts(
+            parentStation.getId(),
+            routeId
+          );
+          if (siblingAlerts != null) {
+            alertsForStopAndRoute.addAll(siblingAlerts);
+          }
         }
+      }
 
-        // ...and siblings - platform may have been changed
-        for (var siblingStop : stop.getParentStation().getChildStops()) {
-          if (!stop.getId().equals(siblingStop.getId())) {
-            Collection<TransitAlert> siblingAlerts = transitAlertService.getStopAndRouteAlerts(
-              stop.getParentStation().getId(),
-              routeId
-            );
-            if (siblingAlerts != null) {
-              alertsForStopAndRoute.addAll(siblingAlerts);
-            }
-          }
+      // Also check multimodal parent
+      MultiModalStation multiModalStation = getMultiModalStation.apply(parentStation);
+      if (multiModalStation != null) {
+        Collection<TransitAlert> multimodalStopAlerts = transitAlertService.getStopAndRouteAlerts(
+          multiModalStation.getId(),
+          routeId
+        );
+        if (multimodalStopAlerts != null) {
+          alertsForStopAndRoute.addAll(multimodalStopAlerts);
         }
       }
-      // TODO SIRI: Add support for fetching alerts attached to MultiModal-stops
```

Comment:
Do you know why they were commented out in the first place?

Reply:
The multimodal stops were not implemented in OTP2, when this code was ported over from OTP1